### ISSUE TITLE
\jlinl is fragile

### ISSUE THREAD

**Issue by phipsgabler** [wg030/jlcode#27]
```latex
\jlinl{Vector{Int}}
```

gets rendered as `ntVectorI`, which is a weird effect.  (I know that listings' inline verbatim command is sensitive, it surprised me that `\jlinl` worked so well from the beginning).  This can be prevented by escaping the braces:

```latex
\jlinl{Vector\{Int\}}
```

but there should be a warning or something.

**Comment by phipsgabler** [comment]
The same holds for SSA variables: `\jlinl{%i}` is still tokenized as a comment, not finishing the group.

**Comment by wg030** [comment]
Thanks for all your reporting, @phipsgabler 
I have no clue what happens here yet, but I will have a close look at it when updating the package at around the end of the month.

**Comment by phipsgabler** [comment]
I guess it has to do with `\lstinline` not liking being wrapped, since it's really a verbatim command.  Let's look at [the code](https://github.com/wg030/jlcode/blob/3e8d45e584dfbbbbdc9c6350611824ce074093a8/jlcode.sty#L1244):

```latex
\newcommand{\jlinl}[1]{%
\lstinline[language=julia,style=jlcodestyle,
basicstyle=\jlinlfont,basewidth=\jlinlem]^^a7#1^^a7}
```

I remember having problemes with that kind of construction long time ago.  What are the `^^a7` tokens, BTW?

Maybe implementing `\jlinl` by `\lstMakeShortInline` helps?

**Comment by wg030** [comment]
> Maybe implementing \jlinl by \lstMakeShortInline helps?

Might be, definitley worth checking it out.

> What are the `^^a7` tokens, BTW?

Well, `\lstinline` requires a delimiting character for the code, which can be choosen by the user. Usually `!` is used for that job. However I wanted to use a character that is not used in Julia either, in order to avoid trouble since otherwise code with `!` could not be displayed correctly. I decided to go for `§`, but it do not work, when using it directly. Hence I decoded the `§` as `^^a7`, which seemed to work out without causing trouble (see [this article](https://agiletribe.wordpress.com/2015/04/07/adding-unicode-characters-to-latex-documents/))
